src/ralph/lib/permissions/views.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+
+from django.contrib.auth import get_user_model
+from django.contrib.auth.models import Permission
+from django.contrib.contenttypes.models import ContentType
+from django.http import HttpResponseForbidden
+
+from ralph.admin.sites import ralph_site
+
+_permission_views = []
+
+
+class PermissionViewMetaClass(type):
+
+    def __new__(cls, name, bases, attrs):
+        codename = 'can_view_{}'.format(name.lower())
+
+        attrs['permision_codename'] = codename
+        new_class = super().__new__(cls, name, bases, attrs)
+        dispatch = getattr(new_class, 'dispatch', None)
+
+        def new_dispatch(self, request, *args, **kwargs):
```

Comment:
first of all, this looks just like decorator - you could use mixin class to do this, instead of metaclass. Second of all - what if someone overwrite dispatch without calling super? It's not super-safe :)